couch/ci/couch.rake

```diff
@@ -20,36 +57,25 @@ namespace :ci do
 
     task :install do
       Rake::Task['ci:common:install'].invoke('couch')
-      sh %(docker run -p #{container_port}:#{container_port} --name #{container_name} -d couchdb:#{couch_version})
+      sh %(docker run -p #{container_port}:#{container_port}\
+        --name #{container_name} -d #{provider.docker_image} #{provider.docker_args})
     end
 
     task before_script: ['ci:common:before_script'] do
-      Wait.for container_port
-      count = 0
-      logs = `docker logs dd-test-couch 2>&1`
-      puts 'Waiting for couchdb to come up'
-      until count == 20 || logs.include?('CouchDB has started')
-        sleep_for 2
-        logs = `docker logs dd-test-couch 2>&1`
-        count += 1
-      end
-      puts 'couchdb is up!' if logs.include? 'CouchDB has started'
-      # Create a test database
-      sh %(curl -X PUT http://localhost:5984/kennel)
+      Wait.for 'http://localhost:5984', 30
+      sleep_for 5 if provider.version == '2'
```

Comment:
Can you modify the log testing statement to work with couch2?

It's a lot more robust than a simple sleep statement. These tests have run thousands of times, and the variability in how long it takes them to come up is pretty high. It took a long time to work some of these kinks out when we switched to docker infrastructure and this was the best solution we found.

I'd be really happy to help you with this if you need it!

We even have a [helper function](https://github.com/DataDog/datadog-sdk-testing/blob/157d2c3380a31c1b03eb6fedbdd69db9d683b19c/lib/tasks/ci/common.rb#L38-L55) loaded into the rake task, `wait_on_docker_logs(container_name, max_wait, *include_array)`. You can see it used [here](https://github.com/DataDog/integrations-core/blob/12abb127ccd1a7cdb686c6b55541db6176a4ae92/cassandra_nodetool/ci/cassandra_nodetool.rake#L47).

I think we wrote this test before we wrote the function and then never went back to use it here.

Reply:
I've tried that before as I didn't like the `sleep_for 5` solution, but it is somehow weird as the database responds, the logs show the latest possible sentence indicating that the cluster is ready but if we don't wait a little longer, the `_stats` endpoint returns an empty array (meaning possibly that they aren't yet computed)

I'm not sure how to work around that TBH except for testing the `_stats` endpoint until it returns something different from an empty array. How does it sounds?

Reply:
That would also be fine! Logs isn't the only solution that works!

Reply:
(To make sure it won't raise an error, you should probably wrap it in a try/catch.)